apps/engineeringblog/utils/index.ts

```diff
@@ -0,0 +1,56 @@
+import fs from "fs";
+import path from "path";
+import matter from "gray-matter";
+import { format } from "date-fns";
+export type Article = {
+  slug: string;
+  title: string;
+  description: string;
+  date: string;
+  featuredImage: string;
+  content: string;
+};
+
+export type ArticleWithoutContent = Omit<Article, "content">;
+
+export function getAllPosts(): ArticleWithoutContent[] {
```

Comment:
Not all contents will be articles/blogs, some would be just normal pages e.g. about. We need to name these functions accordingly.